Выполнены четыре задания ко второму уроку

DenGolubev opened the pull request:
Задания ко второму уроку

Commit:
Выполнены четыре задания ко второму уроку

DenGolubev committed:
Убрал лишний using

DenGolubev committed:
Сделал проверку на ноль в упражнении на проверку четности (Упражнение №3)

DenGolubev committed:
Сделал проверку на ввод в Упражнении 1

Fastaf-master commented on Lesson-2/Min_Max_Temp.cs:
```diff
@@ -15,9 +15,41 @@ public double aver_Temp()
+            string Temp1 = Console.ReadLine();
+            foreach(char a in Temp1)
+            {
+                if(a < '0' || a > '9')
```
А как ввести 15,6?

DenGolubev replied:
Это проверка на ввод чисел

DenGolubev replied:
Все поправил, стало красиво, но не могу обойти goto((((

Fastaf-master commented on Lesson-2/Min_Max_Temp.cs:
```diff
@@ -15,9 +15,41 @@ public double aver_Temp()
+                if(a < '0' || a > '9')
+                {
+                    Console.WriteLine("Вы ввели буквы вместо цифр при вводе минимальной температуры, попробуйте еще раз\n");
+                    Temp1 = Console.ReadLine();
```
Нельзя изменять коллекции по которым ходит foreach

Fastaf-master commented on Lesson-2/Min_Max_Temp.cs:
```diff
@@ -15,9 +15,41 @@ public double aver_Temp()
             Console.WriteLine("Введите максимальную суточную температуру:");
-            max_Temp = Convert.ToDouble(Console.ReadLine());
+            string Temp2 = Console.ReadLine();
+            foreach (char a in Temp2)
```
Дублирование кода. Лучше сделать 1 метод для валидации. Он будет возвращать или double или выкидывать исключение, либо возвращать bool параметр - успешно распарсилось или нет, а результат возвращать через out параметр при успехе

Fastaf-master commented on Lesson-2/Even_number.cs:
```diff
@@ -11,7 +11,11 @@ public bool even_number()
+            if (Number == 0) 
+            {
+                Console.WriteLine("Вы ввели ноль\n");
+                even_number();
```
Если хотите сделать решение с рекурсией, то нужно делать return even_number - на консоли будет все красиво, а вот метод при текущей реализации будет возвращать false при нуле.

Fastaf-master commented on Lesson-2/Even_number.cs:
```diff
@@ -0,0 +1,25 @@
+{
+    class Even_number
+    {
+        private int Number { get; set; }
```
Number используется только в одном методе. Предлагаю его туда и занести

DenGolubev committed:
Немного исправил, char.IsLetter в Min_Max_Temp

Fastaf-master commented on Lesson-2/Even_number.cs:
```diff
@@ -0,0 +1,25 @@
+    class Even_number
+    {
+        private int Number { get; set; }
+        public bool Even_parity { get; private set; }
```
Добавление этого поля достаточно спорная идея. Я бы сделал весь класс статичным и оставил бы в нем один статичный метод.

Fastaf-master commented on Lesson-2/Serial_number_month.cs:
```diff
@@ -0,0 +1,69 @@
+            string month_name = null;
+            switch (n)
+            {
+                case 1:
```
Вот тут enum мог бы сделать код изящнее

DenGolubev replied:
Сделал на Dictionary, правда не проходили еще

Fastaf-master commented on Lesson-2/Program.cs:
```diff
@@ -0,0 +1,46 @@
+        static void Main(string[] args)
+        {
+            Console.WriteLine("Упражнение 1\n");
+            Min_Max_Temp aver_Temp = new Min_Max_Temp();
```
Общее замечание по наименованию. Обычно нижнее подчеркивание не используют в наименовании. Публичные поля и методы классов именуют в большой буквы. Каждое новое слово тоже пишется с большой буквы. Например AverageTemp();

Fastaf-master commented on Lesson-2/Program.cs:
```diff
@@ -0,0 +1,46 @@
+        {
+            Console.WriteLine("Упражнение 1\n");
+            Min_Max_Temp aver_Temp = new Min_Max_Temp();
+            aver_Temp.average_Temp();
```
Странное решение с точки зрения архитектуры. Я бы разделил логику ввода-вывода на консоль и логику вычисления и не смешивал бы ее в одном классе.

DenGolubev committed:
Создал Partial class

DenGolubev committed:
Переименовал

DenGolubev committed:
Включил в проект нужные файлы

DenGolubev committed:
Пока исправил Упражнение 1, использую goto, а это плохо. Не понимаю как обойти.

DenGolubev committed:
Упражнение 2 работа над ошибками.
ДОбавил partial class SerialNumberMonth.Designer, выбор месяца реализовал через Dictionary

DenGolubev committed:
убрал лишний using

DenGolubev committed:
Не законченные исправления к Упражнению 3

DenGolubev committed:
Работу над ошибками - выполнил

DenGolubev committed:
Merge pull request #3 from DenGolubev/correction_of_mistakes

Работа над ошибками
